src/main/java/de/qabel/desktop/daemon/sync/worker/index/memory/InMemorySyncIndex.java

```diff
@@ -0,0 +1,33 @@
+package de.qabel.desktop.daemon.sync.worker.index.memory;
+
+
+import de.qabel.desktop.daemon.sync.worker.index.SyncIndex;
+import de.qabel.desktop.daemon.sync.worker.index.SyncIndexEntry;
+import de.qabel.desktop.nio.boxfs.BoxFileSystem;
+import de.qabel.desktop.nio.boxfs.BoxPath;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class InMemorySyncIndex implements SyncIndex {
+    private final Map<String, SyncIndexEntry> index = new HashMap<>();
+
+    @Override
+    public SyncIndexEntry get(BoxPath relativePath) {
+        if (relativePath.isAbsolute()) {
+            relativePath = BoxFileSystem.getRoot().relativize(relativePath);
+        }
+        String key = relativePath.toString();
+        synchronized (index) {
+            if (!index.containsKey(key)) {
```

Comment:
If you need this pattern often, consider implementing a quick DefaultHashMap that has an embedded factory which you can then just call with `index.get(key, relativePath)`. I think that is cleaner than putting that logic in there, too.